fire/io/bernese/bernese.py

```diff
@@ -0,0 +1,204 @@
+        if crd_fil is not None:
+            self.crd_parse(open(crd_fil, "r").readlines())
+        else:
+            raise Exception("CRD fil skal angives!")
+
+        # Dernæst tilføj kovariansmatricer fra COV-fil hvis angivet
+        if cov_fil is not None:
+            self.cov_parse(open(cov_fil, "r").readlines())
+
+        # Endelig smages til med krydderier fra ADDNEQ
+        if addneq_fil is not None:
+            self.addneq_parse(open(addneq_fil, "r").readlines())
+        else:
+            raise Exception("ADDNEQ fil skal angives!")
+
+    def crd_parse(self, crd_data):
+        """
+        Parse header og datalinjerne fra en koordinat (CRD) fil og opret stationer
+        """
+        linjenr = 0
+        for line in crd_data:
+            linjenr = linjenr + 1
+            if linjenr == 1:  # header med ugenr/comb week
+                self.resultatset.gnss_week = self.crd_parse_weekline(line)
+            elif linjenr == 3:  # header med datum og epoke
+                headerlinje = self.crd_parse_epochdatumline(line)
+                self.resultatset.datum = headerlinje['DATUM']
+                self.resultatset.epoch = strptime(headerlinje['EPOCH'], '%Y-%m-%d %H:%M:%S')
+            elif linjenr >= 7:  # data begynder på linje 7 i CRD-filer
+                if not line.isspace():  # skip evt. tomme linjer
+                    datalinje = self.crd_parse_dataline(line)
+                    station = Station(navn=datalinje['STATION NAME'],
+                                      koordinat=Koordinat(x=datalinje['X'], y=datalinje['Y'], z=datalinje['Z']),
+                                      kovarians=None, spredning=None)  # kovarians of spredning bliver sat senere
+                    self.resultatset.stations[datalinje['STATION NAME']] = station
+
+    def crd_parse_weekline(self, line):
+        """
+        Parse GNSS-ugenummer fra CRD
+        """
+        params = line.split()
+        return int(params[2])
+
+    def crd_parse_epochdatumline(self, line):
+        """
+        Parse datum og epoke
+        """
+        params = line.split()
+        datum = params[3]
+        epoch = params[5] + " " + params[6]
+        return dict(zip(['DATUM', 'EPOCH'], [datum, epoch]))
+
+    def crd_parse_dataline(self, line):
+        params = line.split()
+        return dict(zip(['NR', 'STATION NAME', 'X', 'Y', 'Z', 'FLAG'], params))
+
+    def cov_parse(self, cov_data):
+        """
+        Parse datalinjerne fra en kovarians (COV) fil.
+        """
+        result: dict = {}
+        linjenr = 0
+        temptable = {}
+        for station in self.resultatset.stations.keys():
+            temptable[station] = {}  # opret en tabel med navne fra CRD parsing
+        for line in cov_data:
+            linjenr = linjenr + 1
+            if linjenr == 1:
+                if self.cov_parse_weekline(line) != self.resultatset.gnss_week:
+                    raise Exception("Forskellige GNSS uger i CRD og COV!")
```

Comment:
Brug så specifikke exceptions som muligt, så undgår du at gemme andre fejl væk nedstrøms i koden.
```suggestion
                    raise ValueError("Forskellige GNSS uger i CRD og COV!")
```

Reply:
Done :-)